src/librustdoc/html/render/print_item.rs

```diff
@@ -1491,22 +1491,23 @@ fn item_constant(w: &mut Buffer, cx: &mut Context<'_>, it: &clean::Item, c: &cle
         let value = c.value(tcx);
         let is_literal = c.is_literal(tcx);
         let expr = c.expr(tcx);
-        if value.is_some() || is_literal {
+        if is_literal {
             write!(w, " = {expr};", expr = Escape(&expr));
+        } else if let Some(ref value) = value {
+            write!(w, " = {value};", value = Escape(value));
         } else {
             w.write_str(";");
         }
 
-        if !is_literal {
-            if let Some(value) = &value {
-                let value_lowercase = value.to_lowercase();
-                let expr_lowercase = expr.to_lowercase();
+        let value_lowercase = value.as_ref().map(|s| s.to_lowercase());
```

Comment:
In your earlier PR I meant to suggest removing the *entire* branch including the body (the `//` business) not just the if-condition, sorry if that wasn't clear. This here still prints `3i32 // 3i32` for `1 + 2` if I'm not mistaken.

Reply:
Correct that makes sense, sorry if I might not have understood it earlier. 

```
13: @!hasraw check failed
	`PATTERN` did not match
	// @!hasraw show_const_contents/constant.CONST_I32_HEX.html ';'
50: @hasraw check failed
	`PATTERN` did not match
	// @hasraw show_const_contents/constant.PI.html '= 3.14159265358979323846264338327950288f32;'
51: @hasraw check failed
	`PATTERN` did not match
	// @hasraw show_const_contents/constant.PI.html '; // 3.14159274f32'

```
These tests still seem to be failing, should the code be suggesting something else, I am a bit confused on what the correct comment should be here